### DISCUSSION TITLE
Why en_core_web_trf use spacy's own tokenizer.

### DISCUSSION THREAD

**Discussion by Boltzmachine** [explosion/spaCy#7890] [Help: Coding & Implementations] [answered]
en_core_web_trf use roberta as its base. But when `doc = nlp(text)`, I got spacy'own tokenizer's result. But roberta has its own tokenizer. So how does spacy deal with that? Does spacy change it internally or just use its own tokenizer as inputs of bert. Can I trust the result?

**Comment by polm** [comment, marked as answer]
spaCy uses the Transformers tokenizer internally and aligns it to the spaCy tokenizer results. You can see how this is implemented in [spacy-transformers](https://github.com/explosion/spacy-transformers). 

Because sentencepiece/wordpiece tokenization is generally kind of weird looking, using an alignment to a more traditional tokenizer is pretty common, and spaCy isn't unique in doing this kind of token mapping. If you're doubtful I'd encourage you to compare the results to other packages.

**Reply by Boltzmachine** [reply]
Thanks!
